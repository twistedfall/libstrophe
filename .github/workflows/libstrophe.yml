name: libstrophe
on:
  push:
    branches:
      - "*"
    tags_ignore:
      - "*"
  pull_request:
    branches:
      - "*"
jobs:
  ci:
    strategy:
      fail-fast: false
      matrix:
        libstrophe-version:
          - 0.9.3
          - 0.10.1
          - 0.11.0
          - 0.12.3
          - 0.13.1
          - 0.14.0
    runs-on: ubuntu-24.04
    env:
      LIBSTROPHE_VERSION: ${{ matrix.libstrophe-version }}
      SCCACHE_GHA_ENABLED: "true"
      RUSTC_WRAPPER: "sccache"
    steps:
      - uses: actions/checkout@v4
      - uses: mozilla-actions/sccache-action@v0.0.9
      - uses: dtolnay/rust-toolchain@stable
        with:
          components: clippy,rustfmt

      - run: ci/install.sh
        shell: bash

      - run: ci/script.sh
        shell: bash

  msrv:
    runs-on: ubuntu-24.04
    env:
      SCCACHE_GHA_ENABLED: "true"
      RUSTC_WRAPPER: "sccache"
    steps:
      - uses: actions/checkout@v4
      - uses: mozilla-actions/sccache-action@v0.0.9
      - uses: dtolnay/rust-toolchain@master
        with:
          toolchain: ${{ steps.metadata.outputs.msrv }}
          components: clippy,rustfmt

      - name: Get MSRV package metadata
        id: metadata
        run: cargo metadata --no-deps --format-version 1 | jq -r '"msrv="+(.packages[]|select(.name=="libstrophe").rust_version)' >> $GITHUB_OUTPUT

      - run: ci/checks.sh
        shell: bash
