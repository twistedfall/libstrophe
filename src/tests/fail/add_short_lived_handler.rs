use libstrophe::{Connection, Context, Stanza, StreamError};

fn main() {
	let handler = |_: &Context, conn: &mut Connection, _, _, _: Option<&StreamError>| {
		let handler = |_: &Context, _: &mut Connection, _: &Stanza| { false };
		conn.handler_add(&handler, None, None, None);
	};
	let conn = Connection::new(Context::new_with_null_logger());
	conn.connect_client(None, None, &handler);
}
