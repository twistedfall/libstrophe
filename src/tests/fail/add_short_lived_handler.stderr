error[E0597]: `handler` does not live long enough
 --> $DIR/add_short_lived_handler.rs:6:20
  |
4 |     let handler = |_: &Context, conn: &mut Connection, _, _, _: Option<&StreamError>| {
  |                                 ---- has type `&mut libstrophe::connection::Connection<'1, '_>`
5 |         let handler = |_: &Context, _: &mut Connection, _: &Stanza| { false };
6 |         conn.handler_add(&handler, None, None, None);
  |         -----------------^^^^^^^^-------------------
  |         |                |
  |         |                borrowed value does not live long enough
  |         argument requires that `handler` is borrowed for `'1`
7 |     };
  |     - `handler` dropped here while still borrowed
